### PR TITLE
RR-1157 - Tidy Overview view class

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**

### CODE REVIEW

File: server/views/pages/overview/partials/overviewTab/_actionsHeader.njk
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/_actionsHeader.njk:1]: This was a redundant template from the old Overview design, so I have deleted it

File: server/views/pages/overview/partials/overviewTab/_goalsList.njk
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/_goalsList.njk:1]: This was a redundant template from the old Overview design, so I have deleted it

File: server/views/pages/overview/partials/overviewTab/qualificationsAndAchievementsSidebar.njk
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/qualificationsAndAchievementsSidebar.njk:1]: This was a redundant template from the old Overview design, so I have deleted it

File: server/routes/overview/overviewController.ts [hunk 1]
```diff
@@ -1,92 +1,26 @@
...
-  constructor(
-    private readonly inductionService: InductionService,
-    private readonly educationAndWorkPlanService: EducationAndWorkPlanService,
-  ) {}
+  constructor(private readonly educationAndWorkPlanService: EducationAndWorkPlanService) {}
 
   getOverviewView: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
-    const {
+    const { prisonerSummary, curiousInPrisonCourses, actionPlanReviews, prisonerFunctionalSkills, induction } =
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/overviewController.ts:10]: The controller now gets everything it needs from `res.locals`, which is all populated in the route definition in `index.ts` below

File: server/routes/overview/overviewController.ts [hunk 2]
```diff
@@ -1,92 +1,26 @@
...
-    } = res.locals
-
-    try {
-      const [inductionExists, prisonerGoals] = await Promise.all([
-        this.inductionService.inductionExists(prisonNumber, req.user.username),
-        this.educationAndWorkPlanService.getAllGoalsForPrisoner(prisonNumber, req.user.username),
-      ])
-
-      const isPostInduction = Boolean(inductionExists)
-      const functionalSkills = mostRecentFunctionalSkills(prisonerFunctionalSkills)
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/overviewController.ts:32]: And all this stuff about working out the counts etc has all been moved into the View class below (`OverviewView.ts`) which makes the controller so much simpler 👍

File: server/routes/overview/overviewView.ts [hunk 3]
```diff
@@ -1,23 +1,56 @@
...
-  lastUpdatedDate: Date | null
-  lastUpdatedAtPrisonName: string | null
-  noGoals: boolean
-  goalCounts: { activeCount: number; archivedCount: number; completedCount: number }
-  hasWithdrawnOrInProgressCourses: boolean
-  hasCoursesCompletedMoreThan12MonthsAgo: boolean
-  problemRetrievingData: boolean
-  showServiceOnboardingBanner: boolean
-  actionPlanReviewsCount: string
+  functionalSkills: {
```
Comment (nathanrussell-moj-digital) [inline server/routes/overview/overviewView.ts:17]: There's a lot of change here in the view class, but basically this `RenderArgs` type is the crux of it. This is the spec/definition of the rendered properties in the view 👍 , and all the logic for fulfilling this contract is within this class 👍

File: server/views/pages/overview/partials/overviewTab/_educationAndTrainingSummaryCard.njk [hunk 4]
```diff
@@ -0,0 +1,92 @@
+{# Nunjucks template to display a summary card of the prisoner's eduction and training
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/overviewTab/_educationAndTrainingSummaryCard.njk:1]: I've split the main `overviewTabContents` nunjucks templates into several components now, one for each of the main sections on the page (ie. goals summary card, education & training summary card, and session history summary card)
Each of these sections has it's own template file, with it's own unit test covering all paths 👍 
And each template describes the data that needs to be supplied to it via the view 👍

**Review by lynnette-orr — APPROVED** [review]

**Pull request merged**